### ISSUE TITLE
Proposal: adding dry-run=server (like on kubectl)

### ISSUE THREAD

**Issue by sgendrot** [helm/helm#10335]
Hi,

Helm doesn't have the dry-run=server like kubectl.
So the `helm install/upgrade dry-run=server` can act like `kubectl apply --dry-run=server`

I have 2 usecases:
- the dry-run doesn't output an error if a release name is already use
- a label with a bad value (ex: "my value with spaces"). The dry-run doesn't fail (`kubectl apply --dru-run=server` will fail)

I think, we can catch several errors with this.

**Comment by bacongobbler** [comment]
I believe we discussed this long ago in another ticket. We were OK with the idea. We are just waiting on someone to submit it as a pull request.

One caveat to this is that the current `--dry-run` flag accepts a boolean, whereas you're asking it to change it to match kubectl's expectations (`--dry-run=client` or `--dry-run=server`), which would constitute a breaking change in Helm's API. Perhaps it may be better to implement as a separate feature flag, or hold off until Helm 4.

**Comment by dhruv-chavda-mt** [comment]
Hello, Any updates here?
